### PR TITLE
Add cidr_to_range function to generate a range of IP addresses

### PR DESCRIPTION
**Pull request by rayward**
This excludes the network and broadcast addresses.

```puppet
ips = cidr_to_range('192.168.1.0/29')

# returns this array
[
  '192.168.1.1',
  '192.168.1.2',
  '192.168.1.3',
  '192.168.1.4',
  '192.168.1.5',
  '192.168.1.6',
]
```

### CODE REVIEW

File: spec/functions/cidr_to_range_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+
+describe 'cidr_to_range' do
```
Comment (somic) [inline spec/functions/cidr_to_range_spec.rb]: Could you also please add a test case for /32 cidr to ensure it returns empty list.

**Review by somic — CHANGES_REQUESTED** [review]
Thank you! Just several ruby things.

**Comment by rayward** [comment]
@somic I've implemented the changes as requested!

File: lib/puppet/parser/functions/cidr_to_range.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    ENDHEREDOC
+    ) do |args|
+
+    unless args.length == 1
+      raise Puppet::ParseError, ("cidr_to_range(): wrong number of arguments (#{args.length}; must be 1)")
+    end
+
+    arg = args[0]
+    unless arg.respond_to?('to_s')
```
Comment (somic) [inline lib/puppet/parser/functions/cidr_to_range.rb]: Looks good! Please take this entire "unless" out (pretty much everything in ruby will have :to_s method) and we are good to go.

**Review by somic — CHANGES_REQUESTED** [review]

File: lib/puppet/parser/functions/cidr_to_range.rb (hunk 2 above)
Comment (rayward) [inline lib/puppet/parser/functions/cidr_to_range.rb]: Thanks, yeh that was a bit useless. I've replaced it with a stricter `String` check as any other data type coming into this function would be invalid.

**Comment by somic** [comment]
Thank you!
